### PR TITLE
Enable debugging and more OS options

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
This change does the following.
* Enables qemu logging on the host for create and start options
* Add fedora27 & artful into workloads
* Modify dnf proxy settings to mimic that of Ubuntu

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -411,6 +411,14 @@ the workloads define the following in their cloudinit documents.
 ```
 package_upgrade: {{with .PackageUpgrade}}{{.}}{{else}}false{{end}}
 ```
+The -debug option can be used to follow qemu boot logs by connecting to port
```
Comment (markdryan) [inline README.md]: Could you choose a different name for this option.  The reason is that debug is already defined for the create command and it has a different purpose, to dump out the contents of the cloud-init file.  Now it's overloaded.  What about something like debug-port and have the user pass in the port number?  That way the user can chose a different port if 9999 is already being used.

Comment (markdryan) [reply README.md]: You would need to add this new flag to createFlags as well.

File: ccloudvm.go [hunk 2]
```diff
@@ -233,7 +233,8 @@ func createFlags(ctx context.Context, ws *workspace) (*workload, bool, error) {
 	return wkl, debug, nil
 }
 
-func startFlags(in *VMSpec) error {
+func startFlags(in *VMSpec) (bool, error) {
```
Comment (markdryan) [inline ccloudvm.go]: I sort of feel that it makes sense to include this option in the VMspec structure.  This would allow you to specify the debug port in the workloads but still override it on the command line.  Even better, your choice would be remembered, so the next time you started the VM you wouldn't need to specify the option.

Comment (markdryan) [reply ccloudvm.go]: We could also show the debug port if specified in the status output of the status command, which would be nice.

File: workloads/artful.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
```
Comment (markdryan) [inline workloads/artful.yaml]: Could you split out the new workloads into separate commits?  One commit for both this and the fedora27 workload would be fine.

File: workloads/fedora25.yaml [hunk 4]
```diff
@@ -6,20 +6,25 @@ hostname: singlevm
 ---
 #cloud-config
 write_files:
+{{- if len $.HTTPProxy }}
```
Comment (markdryan) [inline workloads/fedora25.yaml]: Could you split this out into a separate commit?

File: README.md (hunk 1 above)
Comment (ganeshmaharaj) [reply README.md]: Definitely.

**Comment by ganeshmaharaj** [comment]
Splitting this into 3 different ones. Closing this and will open respective ones.
